packages/app/src/cli/models/app/app.ts

```diff
@@ -276,6 +281,17 @@ export class App implements AppInterface {
       }
     }
 
+    const extensionCollections = this.allExtensions.filter(
+      (ext) => ext.isEditorExtensionCollection,
+    ) as ExtensionInstance<EditorExtensionCollectionType>[]
+
+    if (extensionCollections.length > 0) {
+      const errors = validateExtensionsHandlesInCollection(extensionCollections, this.allExtensions)
```

Comment:
Hey Isaac I am wondering what you mean by 

> Is there a reason we want this to happen at deploy time and not at app-load time?

We need to validate this here because we need access to all the extensions in the app to do the validations. This is also similar to what is being done for functions that reference ui extension handles such as [this](https://github.com/Shopify/cli/blob/dc377a363b276f712a1b1d81e0e68ea5f864ec1d/packages/app/src/cli/models/app/app.ts#L278) and I do recall discussing this with you and it was suggested this to be done in `preDeployValidation`. We need these validations [here](https://github.com/Shopify/cli/pull/3728/files#diff-cf7257d7c0c4111afc303b94c039388faf0f27ff4e121ce30f89db3d0c9f2d10R364) because we cannot allow a partner deploy invalid configurations for what should be in the collection.

Reply:
Oh I don't question the necessity of this, what I meant is why do we prefer to this here at `preDeployValidation` and not in the `loader.ts` when we load the extensions.

Reply:
Oh I'd also prefer being consistent because If I were to move this to the loader, we have functions doing validation in preDeployValidation and then we are doing something similar somewhere else.

Reply:
I also see the loader as doing validation on schema not doing validation on specific use cases for each spec

Reply:
That makes sense 👌